### DISCUSSION TITLE
Type hinting support for dynamic model creation

### DISCUSSION THREAD

**Discussion by Dav3whit3** [pydantic/pydantic#4439] [Question] [answered]
Hello guys,

First of all I really like and enjoy your project, cheers :)

I was just wondering if I am doing something wrong when using `pydantic.create_model` as I can't figure out how to properly type the result, and get auto completion with vscode feature _"python.analysis.typeCheckingMode": "basic"_

Example:

I'm getting a response from my db in form of a python dict like this one:
```python
db_response_row = {'profile_id': 1597732,
                   'param_name': 'FOO',
                   'param_value': 'BAR'}
```

and I'm trying to map the field names and types from the first row to use them onto pydantic.create_model like so:

```python
dict_types = { key:(type(val), ...) for key, val in db_response_row.items() }
```

And it is properly working but with not really ideal type/autocompletion results. Am I doing something wrong?

![image](https://user-images.githubusercontent.com/61382175/186724991-70b15801-0d27-40ca-a5b8-43f0c73f7dbe.png)

Full code:

```python
import pydantic

db_response_row = {'profile_id': 1597732,
                   'param_name': 'FOO',
                   'param_value': 'BAR'}
print(db_response_row)


''' Also works with quoted elipsis "..." '''
''' as well as with pydantic.Field(default="") '''
dict_types = {key: (type(val), ...) for key, val in db_response_row.items()}
print(dict_types)

ResponseModel = pydantic.create_model("ResponseModel", **dict_types)
resp = ResponseModel(**db_response_row)

print(resp.profile_id)
print(resp.param_name)
print(resp.param_value)```

**Comment by Dav3whit3** [comment, marked as answer]
I guess it is relted/duplicated as for #[3930](https://github.com/pydantic/pydantic/issues/3930) ??

**Reply by samuelcolvin** [reply]
looks like that's right.

I suspect this is a limitation of mypy/pyright et al., but if there's a way to communicate type information to static type checkers, happy to try.
